### PR TITLE
fix: Resolve issue with script tag compilation in self-hosted Project Settings

### LINKED ISSUE #3378: Script tag in Project Settings doesn't get compiled
I'm trying to install the Google Tag in my project as per the following screen:

project-settings

I suspect there is a bug because when syncing the project using npx webstudio and deploying it with npx vercel, the entire content of the Project Settings is wiped out.

Could you please provide a solution that doesn't involve hacking the script tag into the body tag?

Thanks.

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Corresponding SAAS PR https://github.com/webstudio-is/webstudio-saas/pull/298
(Must be renamed to "release.staging" on release to allow testing)

closes #3378 

## Steps for reproduction

See fixtures now have working scripts.
Publish see saas supports script and language https://rewrites.staging.webstudio.is/builder/f30da574-8022-433e-9a15-050792747b6f .

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

File: packages/cli/src/prebuild.ts [hunk 1]
```diff
@@ -684,27 +713,12 @@ export const prebuild = async (options: {
 
       export const contactEmail = ${JSON.stringify(contactEmail)};
 
+      // @todo: Remove after release
```
Comment (kof) [inline packages/cli/src/prebuild.ts:716]: maybe something more concrete, like issue link

File: packages/cli/src/html-to-jsx.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+          result += escapedValue ? "{`" + escapedValue + "`}" : "";
+        }
+        break;
+      case "element-start":
+        {
+          const tag = convertTagName(walkNode.tagName);
+          result += `<${tag}${attributesToString(walkNode.attributes)}>`;
+        }
+        break;
```
Comment (TrySound) [inline packages/cli/src/html-to-jsx.ts:159]: you can put break inside of block and it will format like if() {}

Comment (istarkov) [reply packages/cli/src/html-to-jsx.ts:157]: yep done

File: fixtures/webstudio-remix-netlify-edge-functions/app/root.tsx [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+import { CustomCode } from "./__generated__/_index";
 
 const Root = () => {
+  // Get language from matches
+  const matches = useMatches();
+
+  const lastMatchWithLanguage = [...matches]
+    .reverse()
+    // @ts-ignore
+    .find((match) => match?.data?.pageMeta?.language != null);
```
Comment (TrySound) [inline fixtures/webstudio-remix-netlify-edge-functions/app/root.tsx:15]: .findLast()?

**Review by kof — APPROVED** [review]

**Pull request merged**